impl/src/main/java/org/ehcache/impl/persistence/DefaultLocalPersistenceService.java

```diff
@@ -89,7 +89,9 @@ private void internalStart() {
         throw new RuntimeException(e);
       }
       try {
-        lock = rw.getChannel().lock();
+        if ((lock = rw.getChannel().tryLock()) == null) {
+          throw new RuntimeException("Couldn't lock rootDir: " + rootDirectory.getAbsolutePath());
```

Comment:
Hey @chrisdennis you're double nesting this exception here.  Maybe you should think about doing this differently?